workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/flows/evolvable/CreateEvolvableTokensFlowHandler.kt

```diff
@@ -11,22 +11,26 @@ import net.corda.core.utilities.unwrap
 
 /** In-line counter-flow to [CreateEvolvableTokensFlow]. */
 class CreateEvolvableTokensFlowHandler(val otherSession: FlowSession) : FlowLogic<Unit>() {
-    @Suspendable
-    override fun call() {
-        // Receive the notification
-        val notification = otherSession.receive<CreateEvolvableTokensFlow.Notification>().unwrap { it }
+	@Suspendable
+	override fun call() {
+		val ourKeys = serviceHub.keyManagementService.filterMyKeys(serviceHub.keyManagementService.keys)
 
-        // Sign the transaction proposal, if required
-        if (notification.signatureRequired) {
-            val signTransactionFlow = object : SignTransactionFlow(otherSession) {
-                override fun checkTransaction(stx: SignedTransaction) = requireThat {
-                    // TODO
-                }
-            }
-            subFlow(signTransactionFlow)
-        }
+		// Receive the notification
+		val notification = otherSession.receive<CreateEvolvableTokensFlow.Notification>().unwrap { it }
 
-        // Resolve the creation transaction.
-        subFlow(ObserverAwareFinalityFlowHandler(otherSession))
-    }
+		// Sign the transaction proposal, if required
+		if (notification.signatureRequired) {
+			val signTransactionFlow = object : SignTransactionFlow(otherSession) {
+				override fun checkTransaction(stx: SignedTransaction) = requireThat {
+					require(stx.getMissingSigners().any { it in ourKeys }) {
```

Comment:
Aren't more checks needed here? This is just checking the counterparty signature has been requested. But malicious node could craft any txn with the counterparty signature in it.

Reply:
The handler now only signs the transaction if it is one of the maintainers in one of the output states of the transaction.
This mirrors the client side which requests signatures from just the maintainers.

Reply:
Updated - removed redundant code. 
Yes the SignTransactionFlow receives the signing key expected to be used. 
Also redundant code was using getMissingSigners, where as code in SignTransactionFlow checks against requiredSigningKeys. (Processing could have been different if using composite keys).